tests/integration_tests/build/test_kani.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Proofs ensuring memory safety properites, user-defined assertions,
+absence of panics and some types of unexpected behavior (e.g., arithmetic overflows).
+"""
+import concurrent.futures
+import platform
+
+import pytest
+
+from framework import utils
+
+PLATFORM = platform.machine()
+CRATES_WITH_PROOFS = ["dumbo", "rate_limiter"]
+
+
+@pytest.mark.timeout(1800)
+@pytest.mark.skipif(PLATFORM != "x86_64", reason="Kani proofs run only on x86_64.")
+def test_kani():
+    """
+    Test all Kani proof harnesses.
+    """
+
+    with concurrent.futures.ThreadPoolExecutor() as executor:
```

Comment:
Do you want to limit the pool somehow? Now it is just 2 threads, but maybe this would grow?

`multiprocessing.cpu_count()` could be a good default.

Reply:
Mh, this is limited by the number of crates we have, and since we're in the process of moving them all into vmm, I don't think this'll ever become a problem. I like the idea of parallelizing using xdist though!